### ISSUE TITLE
Firebase Database plugin (5.4.0, IOS, .Net 3.x) - InitializationException:  Firebase app creation failed.

### ISSUE THREAD

**Issue by studentutu** [firebase/quickstart-unity#239]
When building for IOS with the Latest Google Play Games 0.9.52, Latest Firebase Database (5.4.0), And Latest Admob plugins for Unity, Application crashes when trying to create Firebase Database Reference

Unity 208.14.f1, With Unity Ads, IAP, Google Play Games, Firebase Database, Admob
 ON Mac Mojave with latest Update (as for 01.11.2018),
pod version 1.5.3

Here is a debug from Xcode:
ERROR: Unable to configure Firebase services: Google Analytics for Firebase version (50200000) does not match with Google App Measurement (50300000) version. Please update.
Unable to configure Firebase services: Google Analytics for Firebase version (50200000) does not match with Google App Measurement (50300000) version. Please update.
 
(Filename: /Users/builduser/buildslave/unity/build/Runtime/Export/Debug.bindings.h Line: 43)

InitializationException:  Firebase app creation failed.
  at Firebase.FirebaseApp.CreateAndTrack (Firebase.CreateDelegate createDelegate, Firebase.FirebaseApp existingProxy) [0x00000] in <filename unknown>:0 
  at Firebase.Database.FirebaseDatabase.get_DefaultInstance () [0x00000] in <filename unknown>:0 

Please, help.

**Comment by studentutu** [comment]
I and my College have also encountered the issue with gem and cocoapods, namely, if the previous version of the cocoapods was installed and you try to install it once again (to update), in some weird cases  Unity will not link the Pods to the Xcode project.  The way to fight this issue is to remove the Xcode, then: gem unistall --all (with all of the dependencies), then re-install with:  sudo gem install -n /usr/local/bin cocoapods, and after do : pod setup

Then build the project for the Xcode, and try to build in Xcode.
This indeed helped to make Xcode build, but when launched on the device - there is always  InitializationException: Firebase app creation failed.

**Comment by studentutu** [comment]
Just found out: 
IOS, Mac Mojave, Unity Editor 2018.2.14f1, .Net 3.x

using the Empty project
install only Firebase Database 5.4.0 

Make a simple script to test database

1. There is always a warning that there is no GoogleServices... .plist and googleservices.json, but they are inside the Assets!
2. When you run it inside the editor, now it will give you the error - nof found Assets/StreamingsAssets/ GoogleServices... .plist, Assets/StreamingsAssets/googleservices.json ! Right after you will get an error with  Firebase app creation failed!

**Comment by a-maurice** [comment]
Hi there,

Sorry for the delay in getting back to you.  It seems like there has been a problem with the Cocoapod versions that the Unity SDK is being pinned to.  We'll try to get a new version of the Unity SDK out with a fix as soon as possible.

In the meantime, what you can do is, in the Assets/Firebase/Editor folder, there will be a handful of files of the form *Dependencies.xml (likely AppDependencies.xml, AuthDependencies.xml, and DatabaseDependencies.xml).  In each of those, there would be a tag for iosPod, which will have version="5.9.0".  If you update that to "5.11.0", that should hopefully fix your issue.

Thanks for notifying us of this problem.

**Comment by studentutu** [comment]
@a-maurice  I will try it later today, for now, we have to go back and downgrade the firebase to 5.3.0, it is stable for us

**Comment by studentutu** [comment]
@a-maurice  yes it worked! Thank you!

**Comment by a-maurice** [comment]
Great to hear.  And we have released a new version with it properly fixed, 5.4.1. 
 https://firebase.google.com/support/release-notes/unity#5.4.1

The fix ended up being slightly different, pinning the AppMeasurement cocoapod to 5.2.  The 5.11.0 fix I suggested might work for you, but we did discover a problem with that approach when trying to use Cloud Messaging, so I'd recommend swapping to the official fix if possible.
